### PR TITLE
refactor(core): Revamp subworkflow policy checker errors

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/PAY-1760

### CODE REVIEW

**Comment by ivov** [comment]
e2e: https://github.com/n8n-io/n8n/actions/runs/10791142978

File: packages/cli/src/services/access.service.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Service } from 'typedi';
+import { SharedWorkflowRepository } from '@/databases/repositories/shared-workflow.repository';
+import { UserRepository } from '@/databases/repositories/user.repository';
+import type { User } from '@/databases/entities/user';
+import type { Workflow } from 'n8n-workflow';
+
+@Service()
+export class AccessService {
```
Comment (tomi) [inline packages/cli/src/services/access.service.ts:8]: Description?

File: packages/cli/src/services/access.service.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	constructor(
+		private readonly userRepository: UserRepository,
+		private readonly sharedWorkflowRepository: SharedWorkflowRepository,
+	) {}
+
+	/** Whether the user has access to the workflow via project and scope. */
+	async hasAccess(userId: User['id'], workflowId: Workflow['id']) {
```
Comment (tomi) [inline packages/cli/src/services/access.service.ts:15]: I think this checks specifically read access, right? Maybe this should be called `hasReadAccessToWorkflow`

File: packages/cli/src/subworkflows/__tests__/subworkflow-policy-checker.test.ts [hunk 3]
```diff
@@ -97,68 +76,95 @@ describe('SubworkflowPolicyChecker', () => {
...
+
+			ownershipService.getWorkflowProjectCached.mockResolvedValueOnce(parentWorkflowProject);
+			ownershipService.getWorkflowProjectCached.mockResolvedValueOnce(subworkflowProject);
+
+			const check = checker.check(subworkflow, parentWorkflow.id);
+
+			await expect(check).rejects.toThrowError(SubworkflowPolicyDenialError);
+		});
+
+		it('should not throw', async () => {
```
Comment (tomi) [inline packages/cli/src/subworkflows/__tests__/subworkflow-policy-checker.test.ts:127]: in which case?

File: packages/cli/src/subworkflows/subworkflow-policy-checker.service.ts [hunk 4]
```diff
@@ -30,27 +36,45 @@ export class SubworkflowPolicyChecker {
...
+	private async errorDetails(subworkflowProject: Project, subworkflow: Workflow, userId?: string) {
+		const hasAccess = userId
+			? await this.accessService.hasAccess(userId, subworkflow.id)
+			: false; /* no user ID in policy check for error workflow, so `false` to keep error message generic */
+
+		if (subworkflowProject.type === 'team') return { hasAccess };
+
+		const owner = await this.ownershipService.getPersonalProjectOwnerCached(subworkflowProject.id);
+
+		assert(owner !== null); // only `null` if not personal
```
Comment (tomi) [inline packages/cli/src/subworkflows/subworkflow-policy-checker.service.ts:72]: 👍

**Review by tomi — COMMENTED** [review]
Looking good! 🎉  Only couple suggestions about naming/documentation

File: packages/cli/src/services/access.service.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/services/access.service.ts]: Updated to `hasReadAccess` as workflow is already in the arg and type.

**Review by tomi — APPROVED** [review]
Nice work! 🚀

**Pull request merged**